manifests/plugin/remote_execution/mosquitto.pp

```diff
@@ -20,14 +20,15 @@
 #
 class foreman_proxy::plugin::remote_execution::mosquitto (
   Stdlib::Port $port = 1883,
-  Stdlib::Absolutepath $ssl_ca = $foreman_proxy::ssl_ca,
-  Stdlib::Absolutepath $ssl_cert = $foreman_proxy::ssl_cert,
-  Stdlib::Absolutepath $ssl_key = $foreman_proxy::ssl_key,
+  Stdlib::Absolutepath $ssl_ca = undef,
+  Stdlib::Absolutepath $ssl_cert = undef,
+  Stdlib::Absolutepath $ssl_key = undef,
   Boolean $require_certificate = true,
   Boolean $use_identity_as_username = true,
 ) {
   $mosquitto_config_dir = '/etc/mosquitto'
   $mosquitto_ssl_dir = "${mosquitto_config_dir}/ssl"
+  $broker = "mqtts://${facts['networking']['fqdn']}:${port}"
```

Comment:
Because of this, `$broker` contains protocol, hostname and port. In sp-rex-ssh's settings (where `$broker` is reused), hostname is taken from `mqtt_broker`, port from `mqtt_port` and protocol is determined by availability of certs (or `mqtt_tls`, if set). Surprisingly enough, it seems to still work even if everything lives in `mqtt_broker`.

Maybe it would be better to have `mqtt_broker` be only the hostname to avoid confusion which may come up from things being possible to set in multiple places?

Reply:
I fixed this up.